dask/_expr.py

```diff
@@ -733,3 +757,236 @@ def collect_dependents(expr) -> defaultdict:
+                    annotations_by_type[annot_type].update(
+                        {k: (value(k) if callable(value) else value) for k in layer}
+                    )
+        return annotations_by_type
+
+    def _operands_for_repr(self):
+        return [
+            f"name={self.operand('name')!r}",
+            f"dsk={self.operand('dsk')!r}",
+        ]
+
+    def _tree_repr_lines(self, indent=0, recursive=True):
+        # TODO: This doesn't make sense yet
+        return self._operands_for_repr()
+
+    def __dask_keys__(self):
+        if self.operand("output_keys"):
+            return self.operand("output_keys")
+        dsk = self.operand("dsk")
+        # Note: This will materialize
+        dependencies = dsk.get_all_dependencies()
+        dependents = reverse_dict(dependencies)
+        return [d for d in dependents if not dependents[d] and d in dsk]
+
+    def _layer(self) -> dict:
+        dsk = self.operand("dsk")
+
+        keys = self.operand("output_keys")
+        optimizer = self.operand("low_level_optimizer")
+        if keys is None and optimizer is not None:
+            keys = self.__dask_keys__()
+
+        if (optimizer := self.operand("low_level_optimizer")) is not None:
+            return ensure_dict(optimizer(dsk, keys))
+        return ensure_dict(dsk)
+
+
+class HLGTuple(Expr):
+    _parameters = ["hlgs", "optimizer"]
+
+    def __getitem__(self, other):
+        return self.operands[other]
+
+    def _operands_for_repr(self):
+        return [
+            f"name={self.operand('name')!r}",
+            f"dsk={self.operand('dsk')!r}",
+        ]
+
+    def _tree_repr_lines(self, indent=0, recursive=True):
+        return self._operands_for_repr()
+
+    def _layer(self) -> dict:
+        from dask.highlevelgraph import HighLevelGraph
+
+        graphs, keys = [], []
+        for v in self.operand("hlgs"):
+            if isinstance(v, HLGExpr):
+                dsk = v.operand("dsk")
+            else:
+                # FinalizeCompute
+                dsk = v._layer()
+            if not isinstance(dsk, HighLevelGraph):
+                dsk = HighLevelGraph.from_collections(
+                    str(id(dsk)), dsk, dependencies=()
+                )
+
+            graphs.append(dsk)
+            keys.append(v.__dask_keys__())
+
+        dsk = HighLevelGraph.merge(*graphs)
+        if (optimizer := self.operand("optimizer")) is not None:
+            return ensure_dict(optimizer(dsk, keys))
+        return ensure_dict(dsk)
+
+    def __dask_annotations__(self):
+        annotations_by_type = {}
+        for v in self.operand("hlgs"):
+            if isinstance(v, HLGExpr):
+                annotations_by_type.update(v.__dask_annotations__())
+        return annotations_by_type
+
+    def __dask_keys__(self) -> list:
+        all_keys = []
+        for op in self.operand("hlgs"):
+            all_keys.append(op.__dask_keys__())
+        return all_keys
+
+
+class Tuple(Expr):
```

Comment:
what are the parameters here?

Reply:
whatever expressions you pass in. this doesn't ahve anything custom